araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,346 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
+            the reference channel.
+        """
+
+        # Calculate the time delay between each channel and the reference channel
+        delays = {}
+        for ch_ID in channels_to_csw:
+
+            if ch_ID == reference_ch: 
+                # Delay between the reference channel and itself will be 0
+                delay = 0
+            else: 
+                
+                # Load  the cross correlation for this channel and the reference 
+                xcorr_times, xcorr_volts = wu.tgraph_to_arrays(
+                    self.__get_correlation_function(
+                        ch_ID, reference_ch, wavepacket, applyHilbert=False
+                    )
+                )
+
+                # Identify the `zoom_window` nanosecond window around the 
+                #   reconstructed delay
+                zoomed_indices = np.where(
+                    (np.abs( xcorr_times - reco_delays[ch_ID] )) < zoom_window // 2
+                )[0]
+
+                # Calculate the time of maximum correlation from this
+                #   window of expected signal delay.
+                if len(zoomed_indices) == 0: 
+
+                    # Software triggers are so short, a channel may not have
+                    #   signal during the time window where signal is expected.
+                    #   As a result, `zoom_indices` will have no entries. 
+                    # Just find the time of peak correlation for the full 
+                    #   waveform in this scenario.
+                    delay = xcorr_times[ np.nanargmax(xcorr_volts) ]
+
+                    # If the event can't find the time window to zoom in on
+                    #   and its not a software trigger, warn user
+                    if trig_type != "soft": 
+                        print(
+                            f"Touble calculating csw for {trig_type} event with "
+                            f"channels {ch_ID} and {reference_ch}")
+
+                else: 
+                    delay = xcorr_times[ 
+                        np.nanargmax(xcorr_volts[zoomed_indices]) # index of max xcorr in zoomed array
+                        + zoomed_indices[0] # Adjusted by first zoomed_index
+                    ]
+
+            # AraRoot always compares channels with smaller IDs to channels with 
+            #   larger IDs but we always want to compare to the reference channel.
+            # Correct for this if the current ch_ID is larger than the reference ch_ID
+            if ch_ID > reference_ch: 
+                delay *= -1
+            
+            # Save the calculated arrival delay
+            delays[ch_ID] = delay
+
+        return delays
+
+    def __trim_array(self, times, values, trim):
+        """
+        Trim an array by `trim` values, being careful to avoind trimming out the
+        maximum signal
+
+        Parameters
+        ----------
+        times : Iterable
+        values : Iterable
+        trim : int
```

Comment:
Is there a reason to do this in terms of index and not trimming to a fixed time window where there will be overlap? Maybe it will be more clear below.

Reply:
Yeah I use this in the context of "waveform 1 is 100 elements long and waveform 2 is 95 elements long, so trim waveform 1 by 5 elements" if that makes sense.

Reply:
Oh trimming to maximize overlap makes way more sense. I'll look into that!

Reply:
Done!